.github/workflows/publish.yml

```diff
@@ -0,0 +1,27 @@
+name: Publish
+
+on: push
```

Comment:
what about PRs?

Reply:
As it is now, it will do the build step (`compile.sh`) on the source branch, but only publish on main branch.

I think gh-actions also have specific triggers for PRs, though I've usually found it enough to have it run on the source branch.

Reply:
`push` would only consider the local branches. Typical PRs come from forks, so there is no "push" involved

Reply:
ahh, you're right it would only run in the forked repo, adding pull_request trigger as well